spec/lib/data_pull_spec.rb

```diff
@@ -271,4 +271,74 @@
       end
     end
   end
+
+  describe DataPull::UuidExport do
+    subject(:subtask) { DataPull::UuidExport.new }
+
+    let(:user1) { create(:user, :fully_registered) }
+    let(:user2) { create(:user, :fully_registered) }
+
+    let(:agency) { create(:agency) }
+    let(:service_provider) { create(:service_provider, agency_id: agency.id) }
+    let(:other_sp) { create(:service_provider, active: true, agency_id: agency.id) }
+
+    let(:identity) do
+      create(:service_provider_identity, service_provider: service_provider.issuer, user: user1)
+    end
+    let(:other_identity) do
+      create(:service_provider_identity, service_provider: other_sp.issuer, user: user2)
+    end
+
+    let(:agency_identity) do
+      create(:agency_identity, agency: agency, user: user1, uuid: identity.uuid)
+    end
+    let(:other_agency_identity) do
+      create(:agency_identity, agency: agency, user: user2, uuid: other_identity.uuid)
+    end
+
+    let(:args) { [user1.uuid, user2.uuid, 'does-not-exist'] }
+    let(:include_missing) { true }
+    subject(:result) { subtask.run(args: args, config: config) }
+
+    describe '#run' do
+      context 'without requesting issuers' do
+        let(:config) { ScriptBase::Config.new(include_missing: include_missing) }
+
+        it 'return partner UUIDs for all apps', aggregate_failures: true do
+          expected_table = [
+            ['login_uuid', 'agency', 'issuer', 'external_uuid'],
+            [user1.uuid, agency.name, service_provider.issuer, agency_identity.uuid],
+            [user2.uuid, agency.name, other_sp.issuer, other_agency_identity.uuid],
+            ['does-not-exist', '[NOT FOUND]', '[NOT FOUND]', '[NOT FOUND]'],
+          ]
+
+          expect(result.table).to eq(expected_table)
```

Comment:
I hit some flakiness with this spec today in an unrelated pull request:

https://gitlab.login.gov/lg/identity-idp/-/jobs/651103

Reply:
Doesn't seem like any sort would be guaranteed on the order of how users are listed?

Reply:
https://github.com/18F/identity-idp/pull/8995